Zotero/Controllers/IdleTimerController.swift

```diff
@@ -8,35 +8,96 @@
 
 import Foundation
 import UIKit
+
 import CocoaLumberjackSwift
+import RxSwift
 
 final class IdleTimerController {
+    private static let customIdleTimerTimemout = 1200
+    private let disposeBag: DisposeBag
     /// Processes which require idle timer disabled
     private var activeProcesses: Int = 0
+    private var activeTimer: DispatchSourceTimer?
+
+    init() {
+        disposeBag = DisposeBag()
+        observeLowPowerMode()
+
+        func observeLowPowerMode() {
+            NotificationCenter.default.rx
+                .notification(.NSProcessInfoPowerStateDidChange)
+                .observe(on: MainScheduler.instance)
+                .subscribe(onNext: { [weak self] _ in
+                    guard ProcessInfo.processInfo.isLowPowerModeEnabled, let self else { return }
+                    forceStopIdleTimer()
+                })
+                .disposed(by: disposeBag)
+        }
+    }
+
+    func resetCustomTimer() {
+        inMainThread { [weak self] in
+            guard let activeTimer = self?.activeTimer else { return }
+            DDLogInfo("IdleTimerController: reset idle timer")
+            activeTimer.suspend()
+            activeTimer.schedule(deadline: .now() + DispatchTimeInterval.seconds(Self.customIdleTimerTimemout))
+            activeTimer.resume()
+        }
+    }
 
-    func disable() {
+    func startCustomIdleTimer() {
+        guard !ProcessInfo.processInfo.isLowPowerModeEnabled else { return }
         inMainThread { [weak self] in
-            guard let self = self else { return }
-            self.activeProcesses += 1
-            UIApplication.shared.isIdleTimerDisabled = true
-            DDLogInfo("IdleTimerController: disable idle timer \(self.activeProcesses)")
+            guard let self else { return }
+            activeProcesses += 1
+            DDLogInfo("IdleTimerController: disable idle timer \(activeProcesses)")
+            guard activeTimer == nil else { return }
+            set(disabled: true)
+            startTimer(controller: self)
+        }
+
+        func startTimer(controller: IdleTimerController) {
+            let timer = DispatchSource.makeTimerSource(flags: [], queue: .main)
+            timer.schedule(deadline: .now() + DispatchTimeInterval.seconds(Self.customIdleTimerTimemout))
+            timer.setEventHandler(handler: { [weak controller] in
+                controller?.forceStopIdleTimer()
+            })
+            timer.resume()
+            controller.activeTimer = timer
         }
     }
 
-    func enable() {
+    func stopCustomIdleTimer() {
         inMainThread { [weak self] in
-            guard let self = self else { return }
-            guard self.activeProcesses > 0 else {
+            guard let self, activeProcesses > 0 else {
```

Comment:
If `activeProcesses` is `0`, log a warning but continue to stop the timer.

Reply:
Fixed in ec21879cebda856ecb862bc78281538db6705d31